instruments/tests/test_tektronix/test_tktds5xx.py

```diff
@@ -0,0 +1,869 @@
+                "TRIG:MAI:EDGE:SOU?"
+            ],
+            [
+                f"{source.value}"
+            ]
+    ) as inst:
+        inst.trigger_source = source
+        assert inst.trigger_source == source
+
+
+def test_trigger_source_type_error():
+    """Raise type error when source is not an `source` enum."""
+    wrong_type = 42
+    with expected_protocol(
+            ik.tektronix.TekTDS5xx,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        with pytest.raises(TypeError) as err_info:
+            inst.trigger_source = wrong_type
+        err_msg = err_info.value.args[0]
+        assert err_msg == f"Trigger source setting must be a " \
+                          f"`TekTDS5xx.Trigger` value, got " \
+                          f"{type(wrong_type)} instead."
+
+
+@given(dt=st.datetimes(min_value=datetime(1000, 1, 1)))
+def test_clock(dt):
+    """Get / set oscilloscope clock."""
+    # create a date and time
+    dt_fmt_receive = '"%Y-%m-%d";"%H:%M:%S"'
+    dt_fmt_send = 'DATE "%Y-%m-%d";:TIME "%H:%M:%S"'
+    with expected_protocol(
+            ik.tektronix.TekTDS5xx,
+            [
+                dt.strftime(dt_fmt_send),
+                "DATE?;:TIME?"
+            ],
+            [
+                dt.strftime(dt_fmt_receive)
+            ]
+    ) as inst:
+        inst.clock = dt
+        assert inst.clock == dt.replace(microsecond=0)
+
+
+def test_clock_value_error():
+    """Raise ValueError when not set with datetime object."""
+    wrong_type = 42
+    with expected_protocol(
+            ik.tektronix.TekTDS5xx,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        with pytest.raises(ValueError) as err_info:
+            inst.clock = wrong_type
+        err_msg = err_info.value.args[0]
+        assert err_msg == f"Expected datetime.datetime but got " \
+                          f"{type(wrong_type)} instead"
+
+
+@pytest.mark.parametrize("newval", (True, False))
+def test_display_clock(newval):
+    """Get / set if clock is displayed on screen."""
+    with expected_protocol(
+            ik.tektronix.TekTDS5xx,
+            [
+                f"DISPLAY:CLOCK {int(newval)}",
+                "DISPLAY:CLOCK?"
+            ],
+            [
+                f"{int(newval)}"
+            ]
+    ) as inst:
+        inst.display_clock = newval
+        assert inst.display_clock == newval
+
+
+def test_display_clock_value_error():
+    """Raise ValueError when display_clock is called w/o a bool."""
+    wrong_type = 42
+    with expected_protocol(
+            ik.tektronix.TekTDS5xx,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        with pytest.raises(ValueError) as err_info:
+            inst.display_clock = wrong_type
+        err_msg = err_info.value.args[0]
+        assert err_msg == f"Expected bool but got {type(wrong_type)} instead"
+
+
+def test_get_hardcopy(mocker):
```

Comment:
The `get_hardcopy` routine is weird to me. I can't seem to find it in the programmers manual. Furthermore, looking up header length for `bmp` files, which I think should be transferred here, does not show a "standard" header length of 54, as is assumed here. 
The test here is basically stiched together such that the routine is tested...